lib/rules/no-redeclare.js

```diff
@@ -65,8 +91,112 @@ module.exports = {
 				yield { type: "builtin" };
 			}
 
-			for (const id of variable.identifiers) {
-				yield { type: "syntax", node: id, loc: id.loc };
+			const identifiers = variable.identifiers
+				.map(id => ({
+					identifier: id,
+					parent: id.parent,
+				}))
+				// ignore function declarations because TS will treat them as an overload
+				.filter(({ parent }) => parent.type !== "TSDeclareFunction");
+
+			// Handle TypeScript declaration merging if enabled
+			if (ignoreDeclarationMerge && identifiers.length > 1) {
+				// interfaces merging
+				if (
+					identifiers.every(
+						({ parent }) =>
+							parent.type === "TSInterfaceDeclaration",
+					)
+				) {
+					return;
+				}
```

Comment:
[Refactor] This is handled by the `class + interface/namespace merging` logic since `CLASS_DECLARATION_MERGE_NODES` includes interfaces.

```suggestion
```

Reply:
Done.